app/models/labware_creators/donor_pooling_plate.rb

```diff
@@ -218,7 +238,7 @@ def tag_depth_hash
     def build_pools
       groups = split_single_group_by_study_and_project(source_wells_for_pooling)
       groups = split_groups_by_unique_donor_ids(groups)
-      distribute_groups_across_pools(groups, number_of_pools)
+      distribute_groups_across_pools(groups, calculated_number_of_pools)
```

Comment:
Note that `distribute_groups_across_pools` uses a halving algorithm in distribution at the moment, which causes uneven distribution in small number of pools. I think It needs a weighted sharing of pools to each group based on size first and then either a round robin or a fair sharing of remainder after integer division for distribution in them.